src/traces/bar/cross_trace_calc.js

```diff
@@ -126,9 +126,14 @@ function initBase(gd, pa, sa, calcTraces) {
         // time. But included here for completeness.
         var scalendar = trace.orientation === 'h' ? trace.xcalendar : trace.ycalendar;
 
+        // 'base' on categorical axes makes no sense
+        var d2c = sa.type === 'category' || sa.type === 'multicategory' ?
+            function() { return null; } :
+            sa.d2c;
```

Comment:
Ha clever - not the most performant solution, but possibly the least code, and since this is a failsafe rather than a real use case that's the more important criterion!

Reply:
> that's the more important criterion!

I'm glad you agree!